damnit/ctxsupport/ctxrunner.py

```diff
@@ -384,64 +409,89 @@ def train_timestamps():
+    exec_ap.add_argument('proposal', type=int)
+    exec_ap.add_argument('run', type=int)
+    exec_ap.add_argument('run_data', choices=('raw', 'proc', 'all'))
+    exec_ap.add_argument('--mock', action='store_true')
+    exec_ap.add_argument('--cluster-job', action="store_true")
+    exec_ap.add_argument('--match', action="append", default=[])
+    exec_ap.add_argument('--save', action='append', default=[])
+    exec_ap.add_argument('--save-reduced', action='append', default=[])
+
+    ctx_ap = subparsers.add_parser("ctx", help="Evaluate context file and pickle it to a file")
+    ctx_ap.add_argument("context_file", type=Path)
+    ctx_ap.add_argument("out_file", type=Path)
 
     args = ap.parse_args(argv)
     logging.basicConfig(level=logging.INFO)
 
     db_conn = sqlite3.connect('runs.sqlite', timeout=30)
 
-    # Check if we have proc data
-    proc_available = False
-    if args.mock:
-        # If we want to mock a run, assume it's available
-        proc_available = True
-    else:
-        # Otherwise check with open_run()
-        try:
-            extra_data.open_run(args.proposal, args.run, data="proc")
+    if args.subcmd == "exec":
+        # Check if we have proc data
+        proc_available = False
+        # Check if we have proc data
+        proc_available = False
+        if args.mock:
+            # If we want to mock a run, assume it's available
             proc_available = True
-        except FileNotFoundError:
-            pass
-        except Exception as e:
-            log.warning(f"Error when checking if proc data available: {e}")
-
-    run_data = RunData(args.run_data)
-    if run_data == RunData.ALL and not proc_available:
-        log.warning("Proc data is unavailable, only raw variables will be executed.")
-        run_data = RunData.RAW
-
-    ctx_whole = ContextFile.from_py_file(Path('context.py'), external_vars = get_user_variables(db_conn))
-    ctx = ctx_whole.filter(
-        run_data=run_data, cluster=args.cluster_job, name_matches=args.match
-    )
-    log.info("Using %d variables (of %d) from context file", len(ctx.vars), len(ctx_whole.vars))
-
-    if args.mock:
-        run_dc = mock_run()
-    else:
-        # Make sure that we always select the most data possible, so proc
-        # variables have access to raw data too.
-        actual_run_data = RunData.ALL if run_data == RunData.PROC else run_data
-        run_dc = extra_data.open_run(args.proposal, args.run, data=actual_run_data.value)
-
-    inputs = {
-        'run_data' : run_dc,
-        'db_conn' : db_conn
-    }
-    res = Results.create(ctx, inputs, args.run, args.proposal)
-
-    for path in args.save:
-        res.save_hdf5(path)
-    for path in args.save_reduced:
-        res.save_hdf5(path, reduced_only=True)
+        else:
+            # Otherwise check with open_run()
+            try:
+                extra_data.open_run(args.proposal, args.run, data="proc")
```

Comment:
Not related to this PR, but reading this code, I wonder if we can do this check only when we're going to use the result? When the raw data arrives and DAMNIT starts working on that, calibration is also starting and might have produced some files which this will try to open.

Reply:
Yep, I ran into exactly that recently :stuck_out_tongue: #104 We could avoid it for processing non-`proc` variables, or at least when `args.run_data == 'raw'`.